tests/unit/test_stage_components.py

```diff
@@ -0,0 +1,367 @@
+# pylint: disable=invalid-name
+import json
+from unittest import TestCase
+from datetime import datetime
+import ddt
+import mock
+from xml.etree import ElementTree
+
+from xblock.field_data import DictFieldData
+from xblock.fields import ScopeIds
+from xblock.runtime import Runtime
+
+from group_project_v2.group_project import GroupActivityXBlock
+from group_project_v2.project_api import ProjectAPI
+from group_project_v2.project_navigator import ProjectNavigatorViewXBlockBase
+from group_project_v2.stage import BaseGroupActivityStage
+from group_project_v2.stage_components import StaticContentBaseXBlock, GroupProjectSubmissionXBlock, \
+    GroupProjectReviewQuestionXBlock
+from group_project_v2.upload_file import UploadFile
+from tests.utils import TestWithPatchesMixin, make_api_error
+
+
+class StageComponentXBlockTestBase(TestCase, TestWithPatchesMixin):
+    block_to_test = None
+
+    def setUp(self):
+        super(StageComponentXBlockTestBase, self).setUp()
+        self.runtime_mock = mock.create_autospec(Runtime)
+        self.stage_mock = mock.create_autospec(BaseGroupActivityStage)
+
+        # pylint: disable=not-callable
+        self.block = self.block_to_test(self.runtime_mock, field_data=DictFieldData({}), scope_ids=mock.Mock())
+        self.make_patch(self.block_to_test, 'stage', mock.PropertyMock(return_value=self.stage_mock))
+        self.fragment_class_mock = mock.Mock()
+        patcher = mock.patch('group_project_v2.stage_components.Fragment', self.fragment_class_mock)
+        patcher.start()
+
+        self.addCleanup(patcher.stop)
+
+    # pylint: disable=no-self-use
+    def _assert_empty_fragment(self, fragment):
+        fragment.add_content.assert_not_called()
+        fragment.add_javascript_url.assert_not_called()
+        fragment.add_javascript.assert_not_called()
+        fragment.add_css_url.assert_not_called()
+        fragment.add_css.assert_not_called()
+        fragment.add_resources.assert_not_called()
```

Comment:
@smarnach those are not actual fragments - just mocks, so they won't have content and resources. I'm not sure why I have mocked fragment - probably there were some issues, but I can't recall which ones. So I'll try using actual fragment, but if it is problematic it might remain the same as of now.